src/lib/input/input.ts

```diff
@@ -351,7 +350,26 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
    * Implemented as part of MatFormFieldControl.
    * @docs-private
    */
-  get shouldLabelFloat(): boolean { return this.focused || !this.empty; }
+  get shouldLabelFloat(): boolean {
+    if (this._elementRef.nativeElement.nodeName.toLowerCase() === 'select') {
+      return this.shouldLabelFloatForSelect(this._elementRef.nativeElement);
+    } else {
+      return this.focused || !this.empty;
+    }
+  }
+
+  private shouldLabelFloatForSelect(selectElement:HTMLSelectElement): boolean {
```

Comment:
This method should be prefixed with an underscore (applies to any non-public api)

Reply:
Maybe we should have another class only with the select functionality that extends `MatInput`? Otherwise we'll end up sprinkling `if (isSelect)` checks everywhere.